src/main/java/de/efi23a/bot/discord/api/DiscordBot.java

```diff
@@ -0,0 +1,63 @@
+package de.efi23a.bot.discord.api;
+
+import de.efi23a.bot.discord.api.event.JdaReadyEvent;
+import jakarta.annotation.PreDestroy;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.JDABuilder;
+import net.dv8tion.jda.api.requests.GatewayIntent;
+import net.dv8tion.jda.api.utils.ChunkingFilter;
+import net.dv8tion.jda.api.utils.MemberCachePolicy;
+import net.dv8tion.jda.api.utils.cache.CacheFlag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+
+@Getter
+@Configuration
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class DiscordBot {
+
+  private final ApplicationEventPublisher  applicationEventPublisher;
+  private static final String TOKEN_ENV_VARIABLE = "BOT_TOKEN";
+
+  private JDA jda;
+
+  @SneakyThrows
+  private JDA buildJda() {
+    String botToken = System.getenv(TOKEN_ENV_VARIABLE);
+
+    JDABuilder jdaBuilder = JDABuilder.createDefault(botToken)
+        .disableCache(CacheFlag.ACTIVITY)
+        .setMemberCachePolicy(MemberCachePolicy.ONLINE.or(MemberCachePolicy.VOICE))
+        .setChunkingFilter(ChunkingFilter.NONE)
+        .disableIntents(
+            GatewayIntent.GUILD_WEBHOOKS,
+            GatewayIntent.GUILD_PRESENCES,
+            GatewayIntent.GUILD_MESSAGE_TYPING,
+            GatewayIntent.DIRECT_MESSAGE_TYPING
+        ).enableIntents(GatewayIntent.GUILD_MEMBERS);
+    return jdaBuilder.build().awaitReady();
+  }
+
+  @EventListener
+  public void applicationStarted(ContextRefreshedEvent event) {
```

Comment:
Mir ist der Vorteil vom Warten auf das ContextRefreshedEvent hier nicht ganz ersichtlich.
Was passiert wenn der Context auf eine andere Art refreshed wird?

Reply:
Refresh wird aufgerufen, wenn sich etwas an Spring ändert. Es gibt bestimmte Configurationen,d die Spring neuladen lassen und damit discord neubuilden. Damit könnte man einen Config Reloader einbauen

Reply:
Man kann es auch ändern. Es geht mir eher um das CustomEvent und das automatisierte Registrieren der JDA Events

Reply:
Wir müssten JDA bei jedem Context Refresh  ordentlich neustarten. Damit das klappt müssten wir dann ja so gut wie jedes Feature stateless entwickeln.

Reply:
Was meinst du mit ordentlich neustarten? 
Wir machen alles was mit dem erstellen vom DiscordBot zutun hat in einem JdaReadyEvent, sprich automatisch alle Events registrieren, Presence setzen, (?) daten aus Datenbank laden etc...
Das würde uns ermöglichen den Bot durch eine Refresh Funktion "neuzustarten"

Aber wie gesagt, kann auch sein dass ich falsch liege, dann machen wir es ohne das 'ContextRefreshedEvent'

Reply:
Der AnnotatedEventManager sollte bei den Events schon vieles abnehmen.

Reply:
Ich rede von den JDA Events, du mit jda.addEventListener(...); einzeln registrieren musst.

Reply:
Die können wir ja auch über Beans holen. Aber das SpringEvent ist dafür glaube ich nicht sinnvoll.

Reply:
Geht mir nicht ums Spring Event, dass kann man gerne rausnehmen. Ich meine nur den Rest, also das funktioniert ja auch ohne das Spring Event, nur dass halt dann eben das JdaReadyEvent woanders gefired wird

Reply:
> Wir müssten JDA bei jedem Context Refresh ordentlich neustarten. Damit das klappt müssten wir dann ja so gut wie jedes Feature stateless entwickeln.

Hab nachgeschaut haut man bei der buildJda() method einfach eine if (this.jda != null) { this.jda.shutdownNow(); this.jda = null; } rein ist das Problem behoben